src/application-manager.js

```diff
@@ -0,0 +1,1727 @@
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as EventEmitter from 'events';
+import * as express from 'express';
+import * as bodyParser from 'body-parser';
+import * as fs from 'fs';
+import * as path from 'path';
+
+import * as constants from './lib/constants';
+import { log } from './lib/supervisor-console';
+
+import { validateTargetContracts } from './lib/contracts';
+import { DockerUtils as Docker } from './lib/docker-utils';
+import { LocalModeManager } from './local-mode';
+import * as updateLock from './lib/update-lock';
+import { checkTruthy, checkInt, checkString } from './lib/validation';
+import {
+	ContractViolationError,
+	NotFoundError,
+	InternalInconsistencyError,
```

Comment:
I added a bunch of these `InternalInconsistencyError` in order to check for `null`s that the coffeescript version assumed would not exist, you may want to double check them but in all cases I saw I think it would have errored anyway but with a worse message